simulator/src/main/java/com/hazelcast/simulator/worker/WorkerMessageProcessor.java

```diff
@@ -92,22 +91,18 @@ private boolean isMaster() {
                 public Boolean call() throws Exception {
                     return isOldestMember();
                 }
-            }, 10, TimeUnit.SECONDS).get(TIMEOUT, TimeUnit.SECONDS);
-        } catch (InterruptedException e) {
-            throw new IllegalStateException(e);
-        } catch (ExecutionException e) {
-            throw new IllegalStateException(e);
-        } catch (TimeoutException e) {
+            }, delaySeconds, TimeUnit.SECONDS).get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        } catch (Exception e) {
             throw new IllegalStateException(e);
         }
     }
 
-    private boolean isOldestMember() {
+    boolean isOldestMember() {
```

Comment:
Yes, I found it very hard to figure out what I had to throw into the public methods, which runs through flawless to a certain private method, just to fail in there with a specific exception etc.

Is it this really dodgy to use package private for tests? I've seen a lot of cases.